platforms/core-configuration/declarative-dsl-core/src/main/resources/org/gradle/internal/declarativedsl/parsing/grammar.bnf

```diff
@@ -39,34 +39,28 @@ root ::= statements
 statement ::= (assignment | expression)
 statements ::= (statement semis?)*
 
-assignment ::= directlyAssignableExpression EQ expression {pin=2}
-expression ::= postfixUnaryExpression
+assignment ::= directlyAssignableExpression EQ primaryExpression {pin=2}
+expression ::= primaryExpression postfixUnarySuffixes
 
```

Comment:
I would expect a rule named `expression` to describe the most general form of an expression in the language. Given that `valueArgument` has a different form of expressions now, I would suggest renaming this rule so that the name is more specific to its role.

Reply:
A counter argument is that renaming it makes the comparison/matching to standard Kotlin grammar even harder to do. I.E. I'm a bit thorn about doing or not doing this change...